### ISSUE TITLE
No GUI for clap plugin in Bitwig

### ISSUE THREAD

**Issue by greatest-ape** [greatest-ape/OctaSine#145]
A report has come in of no GUI appearing in Bitwig when using the CLAP plugin.

**Comment by tank-trax** [comment]
I can confirm this, clap shows in Reaper, not in Bitwig

**Comment by TjShredder** [comment]
I can confirm this as well. VST shows the GUI...

**Comment by greatest-ape** [comment]
The latest release should now work on Windows, but unfortunately not yet on macOS.

**Comment by tank-trax** [comment]
still no GUI in Bitwig for the CLAP on my end v8.4 on Linux Mint 21.1

**Comment by greatest-ape** [comment]
Alright, annoying. For some reason Bitwig doesn’t seem to play nice with baseview / iced_baseview.

**Comment by TjShredder** [comment]
You should report it also to Bitwig, they want CLAP to be a selling point and might help you to sort it out...
Its all new and exciting...

**Comment by tank-trax** [comment]
good idea and I agree

**Comment by porzione** [comment]
Maybe related https://github.com/free-audio/clap/issues/323

**Comment by greatest-ape** [comment]
Yeah, that looks related.

**Comment by greatest-ape** [comment]
If any of you have the time, could you please try this build out and see if it works?

[OctaSine-tmp-2024-03-16-macOS.zip](https://github.com/greatest-ape/OctaSine/files/14623301/OctaSine-tmp-2024-03-16-macOS.zip)
[OctaSine-tmp-2024-03-16-Windows.zip](https://github.com/greatest-ape/OctaSine/files/14623302/OctaSine-tmp-2024-03-16-Windows.zip)

**Comment by josb** [comment]
FYI https://nakst.gitlab.io/tutorial/clap-part-3.html shows `.set_parent` not just storing the host window pointer, but also setting it as the plugin window parent. I was going to try this out after figuring out how to do it.

**Comment by josb** [comment]
No window in Bitwig Studio 5.1.6 using your tmp build.

**Comment by greatest-ape** [comment]
Yes, you’re right about open_parented, so it’s just a way to try to make it work with the baseview API.

Nice! Does CPU use drop again when the plugin window is closed/hidden? It was an issue previously that it did not, because the window wasn’t properly cleaned up.

**Comment by tank-trax** [comment]
I can build the latest VST2 but not the Clap as my rust is not recent enough using Debian 12

**Comment by josb** [comment]
Thanks @greatest-ape, for confirming my understanding around `open_parented()`.

I'll look into the CPU behavior soon and will report back. Looking at recent commits, latest iced_baseview is supposed to have addressed this.

**Comment by greatest-ape** [comment]
> I can build the latest VST2 but not the Clap as my rust is not recent enough using Debian 12

Ok! Are you using Debian-packaged Rust? My current rust version policy is latest stable and I will probably not change it. I was happy to be able to get off nightly Rust a while back.

**Comment by greatest-ape** [comment]
> Thanks @greatest-ape, for confirming my understanding around `open_parented()`.
> 
> I'll look into the CPU behavior soon and will report back. Looking at recent commits, latest iced_baseview is supposed to have addressed this.

Ok, thanks!

I’ve switched to a fork of iced_baseview since I don’t want to upgrade to a more recent iced version yet, since it causes various crashes, bugs and performance issues that I don’t want to spend time on ironing out.

**Comment by RustoMCSpit** [comment]
can confirm too lol

**Comment by RustoMCSpit** [comment]
this is still broken on linux mint

**Comment by greatest-ape** [comment]
> this is still broken on linux mint

Ok, even on v0.9.1?

**Comment by RustoMCSpit** [comment]
> > this is still broken on linux mint
> 
> Ok, even on v0.9.1?

yes

**Comment by greatest-ape** [comment]
Ok :(

**Comment by josb** [comment]
Update: OctaSine-v0.9.1-macOS works fine on macOS 15.2 with Bitwig Studio 5.3 Beta 3.

**Comment by greatest-ape** [comment]
> Update: OctaSine-v0.9.1-macOS works fine on macOS 15.2 with Bitwig Studio 5.3 Beta 3.

Great!

**Comment by josb** [comment]
Fwiw,  OctaSine-v0.9.1-macOS still works fine on macOS 15.4.1 with Bitwig Studio 5.3.8. 👍🏻

**Comment by josb** [comment]
Fwiw, OctaSine-v0.9.1-macOS still works fine on macOS 15.7.2 with Bitwig Studio 6.0 Beta 6. 👍🏻
